src/main/java/com/google/graphgeckos/dashboard/storage/DataRepository.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An interface for providing unified I/O to the storage component.
+ */
+public interface DataRepository {
+  /**
+   * Creates a new database entry, with just the metadata of a commit,
+   * having the passed/failed fields empty.
+   *
+   * @param entryData a JSON in plain string format, must have a "commitHash" field.
+   * @throws IllegalArgumentException if entryData is null or doesn't have a
+   *      "commitHash" field.
+   */
+  void createNewEntry(final ParsedGitData entryData);
+
+  /**
+   * Updates an existing database entry, with the passed/failed information for
+   * a particular buildbot. If no entry is associated to the commit hash provided,
+   * creates a new entry from the metadata included in the {@code updateData}.
```

Comment:
{@code ...} isn't used to reference methods, classes, etc. It displays text in code font without interpreting the text as HTML markup or nested javadoc tags. [An article with the list of javadoc tags](https://www.tutorialspoint.com/java/java_documentation.htm#:~:text=Javadoc%20is%20a%20tool%20which,are%20Java%20multi%2Dline%20comments.)

Reply:
Then what is it used for?

Reply:
>{@code ...} isn't used to reference methods

I disagree. For example, consider its usage in the implementation of ArrayList.java from the standard library: https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l604
Could you point out excerpts from [the article](https://www.tutorialspoint.com/java/java_documentation.htm#:~:text=Javadoc%20is%20a%20tool%20which,are%20Java%20multi%2Dline%20comments.) that discourage the usage of `@code` to reference methods?
Could you suggest an alternative to how to reference methods in javadoc?

Reply:
Referencing methods: https://stackoverflow.com/questions/5915992/how-to-reference-a-method-in-javadoc
Referencing fields: https://stackoverflow.com/questions/14796921/how-to-link-javadoc-to-private-field

And while there are alternatives to referencing methods, it isn't explicitly said anywhere that @code annotation is to be used only for code snippets.

Reply:
It is not, but this annotation is intended to be used to represent snippets of code. If this is considered an acceptable way to refer to variables, that's cool! Thanks for the clarification! :)